### PR TITLE
Java and py4 j installation

### PR DESCRIPTION
**Pull request by ZviranOri**
Features for testing with java:
- Basic java installation & compilation of the java gateway script.
- Java gateway abstraction
- Very simple test using the gateway

### CODE REVIEW

File: tests/test_converter.py [hunk 1]
```diff
@@ -1,12 +1,35 @@
...
+        self.gateway = JavaGateway()
+
+    def test_year_month_day_format_conversion_to_java(self):
+        date = "2018-11-07"
+
+        python_format = "%Y-%M-%d"
+        c = PatternConverter()
+        converted_format = c.convert_to_iso_8601(pattern=python_format)
+
+        dt_formatter = self.gateway.jvm.java.time.format.DateTimeFormatter
```
Comment (Iddoyadlin) [inline tests/test_converter.py]: :1st_place_medal:

Comment (Iddoyadlin) [reply tests/test_converter.py]: This is actually always gonna be the java code that we are running, so can be nice if we live the java code to the java process (besides calling a function with this code)

Comment (ZviranOri) [reply tests/test_converter.py]: Hmmmm interesting idea. We can definitely do that. I'll push it together with the fixes I have for the Java server code monitoring. 
On the meanwhile you can just abstact that by extracting a function that, let's say, gets a format and a date str and returns the formatted date str (?)...

Comment (ZviranOri) [reply tests/test_converter.py]: @Iddoyadlin we now have an entry point to the java server which means we can call the methods in that class with the py4j :)
The java class still might need a little work but the example test now works with it.

**Review by Iddoyadlin — APPROVED** [review]

**Pull request merged**